docs: high-level rewrite (#2947)
ansys/pyfluent: doc/source/getting_started/installation.rst

.. _ref_installation:

============
Installation
============

PyFluent Installation
---------------------

PyFluent supports Python 3.9 through Python 3.12 on Windows, Mac OS and Linux.

PyFluent can be installed, along with all its optional dependencies, using:

.. code:: console
   pip install ansys-fluent-core

Development Installation
------------------------
The PyFluent source repository is available on GitHub. You can clone the repository and set up for local
development with the following commands:

.. code:: console

   git clone https://github.com/ansys/pyfluent.git
   cd pyfluent
   pip install pip -U

>    pip install -e

Reviewer: Just checked, this will require the trailing dot.

pip install [options] [-e] <local project path> ...

Reviewer 2: You're right, if trying without the `.` I get the error: "-e option requires 1 argument"

So this should be:
```suggestion
   pip install -e .
```

   python codegen/allapigen.py

Step-by-Step Explanation
~~~~~~~~~~~~~~~~~~~~~~~~

Clone the Repository
++++++++++++++++++++

.. code:: console

   git clone https://github.com/ansys/pyfluent.git
   cd pyfluent

These commands clone the PyFluent repository from GitHub to your local machine and navigate into
the repository directory.

Upgrade Pip
+++++++++++

.. code:: console

   pip install pip -U

This command upgrades pip to the latest version to ensure compatibility with the latest packages.

Install PyFluent in Editable Mode
+++++++++++++++++++++++++++++++++

.. code:: console

>    pip install -e

Reviewer: ```suggestion
   pip install -e .
```

.. _ref_installation:

============
Installation
============

PyFluent Installation
---------------------

PyFluent supports Python 3.9 through Python 3.12 on Windows, Mac OS and Linux.

PyFluent can be installed, along with all its optional dependencies, using:

> .. code::
> 
>    pip install ansys-fluent-core

Reviewer 3: ```suggestion
.. code:: console
   pip install ansys-fluent-core
```

Development Installation
------------------------

The PyFluent source repository is available on GitHub. You can clone the repository using:
.. code:: console

>    git clone https://github.com/ansys/pyfluent.git

Reviewer: Shall we do `cd pyfluent` immediately after this step? User will need to move to the pyfluent directory before executing `python codegen/allapigen.py`.

Reviewer 2: I think the idea in this part of the code is not to provide the full script, this part is just for the "clone repository command"

The codegen section is down below, that is where we need the `cd pyfluent`, or need to let the user know that they need to move into the local PyFluent repository folder before executing codegen

Reviewer 2: Also, I think I preferred providing the full script in one block as we did before:
```
git clone https://github.com/ansys/pyfluent.git
cd pyfluent
pip install pip -U
pip install -e .
python codegen/allapigen.py  # Generates the API files
```
And then we could explain the commands afterwards?

Reviewer: One thing to keep in mind that user will tend to copy and execute the code verbatim from the website, especially when there is a copy button next to the code.

Reviewer 3: Yes, 'cd pyfluent` step should be mentioned here before  executing `python codegen/allapigen.py`

Author: I'll mostly revert this section

Author: It's useful to have the copyable script in a single block. 

We also want each step to be fully explained. I will move all those explanations up into a single block.

I will remove the comment # Generates the API files in favour of the fuller, separate explanation.

Author: TODO create an issue for the outstanding aspects here.

Author: **This section has been fully updated.**

Author: marking resolved but feel free to reopen

> Once you have cloned the repository, run the following commands. Installing with the

Reviewer 2: The sentence "run the following commands" seems a bit misplaced here, as the commands don't follow that sentence, if that makes sense.
```suggestion
Once you have cloned the repository, you can install PyFluent. Installing with the
```

> -e option overwrites the directory in site-packages with a symbolic link to the repository,

Reviewer: In a fresh environment, the directory ansys/fluent/core won't exist, we can say - creates a symbolic link to the repository in site-packages.

Author: Done it

meaning any changes you make to the PyFluent code reflect automatically when you use PyFluent.

.. code:: console

   pip install pip -U
   pip install -e ./pyfluent

The full PyFluent package includes some required API classes that are auto-generated rather than
being maintained under version control, so they are not present amongst the files in your
cloned repository at this point. You run the auto-generation using a script included in the repository,
as specified below. The auto-generation requires an Ansys Fluent installation.

>    python codegen/allapigen.py

Reviewer 3: .. code:: console

   python codegen/allapigen.py

>    python codegen/allapigen.py

Reviewer 3: .. code:: console

   python codegen/allapigen.py

Reviewer: .. code:: console

.. _ref_installation:

============
Installation
============

PyFluent Installation
---------------------

PyFluent supports Python 3.9 through Python 3.12 on Windows, Mac OS and Linux.

PyFluent can be installed, along with all its optional dependencies, using:

.. code:: console
   pip install ansys-fluent-core

Development Installation
------------------------
The PyFluent source repository is available on GitHub. You can clone the repository and set up for local
development with the following commands:

.. code:: console

   git clone https://github.com/ansys/pyfluent.git
   cd pyfluent
   pip install pip -U
   pip install -e
   python codegen/allapigen.py

Step-by-Step Explanation
~~~~~~~~~~~~~~~~~~~~~~~~
1. Clone the Repository:

.. code:: console

   git clone https://github.com/ansys/pyfluent.git
   cd pyfluent

> - These commands clone the PyFluent repository from GitHub to your local machine and navigate into
>   the repository directory.

Reviewer 2: I think it will look better without a bullet point behind it and without tabbing
```suggestion
These commands clone the PyFluent repository from GitHub to your local machine and navigate into
the repository directory.
```
See:
![image](https://github.com/ansys/pyfluent/assets/132297401/c0d57bd6-e0a1-4e29-9faa-3e870ac05b67)

Alternatively, can consider separating into subsections, first one being e.g.
```
Clone the Repository
++++++++++++++++++++
```
